manager/scheduler/nodeinfo.go

```diff
@@ -29,13 +30,14 @@ func newNodeInfo(n *api.Node, tasks map[string]*api.Task, availableResources api
 		Node:  n,
 		Tasks: make(map[string]*api.Task),
 		ActiveTasksCountByService: make(map[string]int),
-		AvailableResources:        availableResources,
+		AvailableResources:        *availableResources.Copy(),
```

Comment:
Let's change `newNodeInfo` to take `*api.Resources` instead of `api.Resources`, so we aren't making an unnecessary shallow copy when we pass the argument into the function.